File: src/test/java/com/indiepost/repository/PostRepositoryTest.java
package com.indiepost.repository;

import com.indiepost.NewIndiepostApplication;
import com.indiepost.dto.post.PostSummaryDto;
import com.indiepost.enums.Types;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit4.SpringRunner;
import org.springframework.test.context.web.WebAppConfiguration;
import org.springframework.transaction.annotation.Transactional;

import javax.inject.Inject;
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(SpringRunner.class)
@SpringBootTest(classes = NewIndiepostApplication.class)
@WebAppConfiguration
@Transactional
public class PostRepositoryTest {

    @Inject
    private PostRepository repository;

    @Test
    public void findByIdsShouldReturnPostsInRequestedOrder() {
        List<Long> postIds = Arrays.asList(5891L, 1679L, 5897L, 5929L);
        List<PostSummaryDto> posts = repository.findByIds(postIds);
        List<Long> resultIds = posts.stream()
                .map(p -> p.getId())
                .collect(Collectors.toList());
        assertThat(postIds).isEqualTo(resultIds);
    }

    @Test
    public void getStatusByIdShouldReturnPostStatusCorrectly() {
        Long postId = 5929L;
        Types.PostStatus status = repository.getStatusById(postId);
        assertThat(status).isEqualTo(Types.PostStatus.PUBLISH);
    }
}

File: src/test/kotlin/com/indiepost/repository/PostRepositoryTest.kt
package com.indiepost.repository

import com.indiepost.IndiepostBackendApplication
import com.indiepost.enums.Types
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.junit.jupiter.SpringExtension
import org.springframework.test.context.web.WebAppConfiguration
import org.springframework.transaction.annotation.Transactional
import java.util.*
import java.util.stream.Collectors
import javax.inject.Inject

@ExtendWith(SpringExtension::class)
@SpringBootTest(classes = [IndiepostBackendApplication::class])
@WebAppConfiguration
@Transactional
class PostRepositoryTest {

    @Inject
    private lateinit var repository: PostRepository

    @Test
    fun findByIdsShouldReturnPostsInRequestedOrder() {
        val postIds = Arrays.asList(5891L, 1679L, 5929L)
        val posts = repository.findByIds(postIds)
        val resultIds = posts.stream()
                .map<Long> { p -> p.id }
                .collect(Collectors.toList())
        assertThat(postIds).isEqualTo(resultIds)
    }

    @Test
    fun getStatusByIdShouldReturnPostStatusCorrectly() {
        val postId = 5929L
        val status = repository.getStatusById(postId)
        assertThat(status).isEqualTo(Types.PostStatus.PUBLISH)
    }
}
